src/components/Login/Login.js

```diff
@@ -0,0 +1,108 @@
+import { getUser } from '../../modules/user/reducer';
+import './style.css';
+
+
+class Login extends Component{
+    constructor(props) {
+        super()
+        this.state =  {
+            data: {
+                email:'',
+                password:''
+            },
+            check:{
+                email:'',
+                password:''
+            },
+            status: {
+                email:'',
+                password:''
+            }
+        }
+    }
+
+    handleChangeInput = ({ target: { name, value } }) => {
+        this.setState({ data: { ...this.state.data, [name]: value } });
+      };
+      
+    checkEmail = async (e) => {
+        const { email } = this.state.data;
+        const { fetchUserDataSuccess, fetchUserDataFailure } = this.props;
+        const { status: result, user } = await getUserByEmail({ email });
+      
+        (result) ? fetchUserDataSuccess(user) : fetchUserDataFailure(user);
+        this.setState({
+          status: { ...this.state.status, email: result },
+          check: { ...this.state.check, email: true },
+        });
+        this.checkPassword();
+      };
+      
+    checkPassword = async (e) => {
+        const { email, password } = this.state.data;
+        const response = await isPassValid({ email, password });
+        this.setState({
+          status: { ...this.state.status, password: response },
+          check: { ...this.state.check, password: true },
+        });
+      };
+      
+    handleContinue = async (e) => {
+        const { check, status } = this.state;
+        const { user, user:{ id }, userAuth } = this.props;
+        if (check.email && check.password && status.email && status.password) {
+          e.preventDefault();
+          const result = await singInById({ id });
+          const path = result ? '/': 'login';
+          user.isAuth = true;
+          await userAuth(user);
+          this.props.history.push(path);
+        }
+      };
+      
+
+
+    render(){
+        const { check, status } = this.state;
+        return (
+            <>
+            <form className='login-form'>
+                <p className='login-form-title'>Login</p> 
+
+                <label className='login-form-label'>Email</label> 
+                <input className="login-form-input" type="text" name='email' placeholder='Example@gmail.com' onChange={this.handleChangeInput} onBlur={this.checkEmail} required/>
+                {check.email ?
+                        <div className = 'form-status-icon' > 
+                            {status.email ?  <i className="fas fa-check green"></i > : < i className = "fas fa-times red" > </i>} 
+                        </div> :
+                    null
+                }
+
+                <label className='login-form-label'>Password</label>
+                <input className="login-form-input" type="password" name='password' onChange={this.handleChangeInput} onBlur={this.checkPassword.bind(this)} />
```

Comment:
`checkPassword` is an arrow function, you don't need to bind it. And even if you need to bind anything you'd better do it in constructor

Reply:
still relevant